splash/browser_tab.py

```diff
@@ -637,6 +638,26 @@ def runjs(self, js_source, handle_errors=True):
             result_protection=False
         )
 
+    def with_timeout(self, run_coro, timeout, callback, errback):
```

Comment:
1. I'm not sure this API makes sense for BrowserTab; BrowserTab should be usable without Lua - there should be a possibility to add e.g. an alternative Python scripting engine, and BrowserTab is already used by Python code (in portia if I recll correctly). `run_coro` argument looks Lua-specific; we should either find a better name or refactor the code and move Lua-specific features to qtrender_lua.
2. The method needs a docstring.